keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-+ COSI central controller sees a new `BucketAccessRequest` (BAR) and gets the `BAR.BucketAccessClass` (BAC).
-+ COSI creates the cluster-scoped `BucketAccess` (BA) filling in the field from the BAR and BAC.
-+ The sidecar sees the new BA and gRPC calls the driver to grant access to the backend bucket.
-+ COSI updates the BAR and BA to reference each other.
-+ The cosi-node-adapter, which has been waiting for the BAR to be processed, writes access tokens to the app pod's specified mount point.
+The sidecar sets `accessGranted=true` in the BucketAccess object and sets the `credentialsSecret` field to the name and namespace of the generated secret. At this point, the BucketAccess is ready to be utilized by workloads.
 
-## Revoke Bucket Access
-This workflow describes the automation supporting revoking access to an existing backend bucket, and the deletion of the cluster-scoped `BucketAccess` instance.
+## Bucket Provisioning (for pods)
 
-Here is the workflow:
+The following stakeholders are involved in the lifecycle of provisioning bucket for pods:
 
-+ COSI central controller sees the `BucketAccessRequest` (BAR) has been deleted.
-+ COSI gets the `BAR.BucketAccess` (BA).
-+ COSI deletes the BA, but finalizers prevent it from being garbage collected.
-+ The sidecar sees the BA has been deleted and gRPC calls the driver to revoke access to the backend bucket.
-+ COSI removes the BAR and BA finalizers and these instances are garbage collected.
+ - Users  - specify bucket in the pod definition
 
-In addition, the cosi-node-adapter sees the app pod is terminating:
-+ cosi-node-adapter receives _NodeUnstageVolume_ request from kubelet.
-+ adapter removes its finalizer from both the BAR.BA and the BAR.BR.
+Once a valid BucketAccess is available, pods can use it to access buckets.
 
-## Delete BucketAccess 
-The above workflows are triggered by the user. Now we cover worflows triggerd by the admin.
-The most common scenario is likely the case where tokens are compromised and the admin needs to stop their use. In this case the admin may terminate the app pod(s) and delete the `BucketAccess` instances.
+###### 1.  User creates pod with volume of type objectstorage.k8s.io
 
-Here is the workflow:
+A BucketAccess with `accessGranted=true` holds information for accessing buckets. 
 
-+ Admin deletes a `BucketAccess` (BA) which is deferred until the finalizer has been removed.
-+ The sidecar detects the delete and calls the driver to revoke access to the backend bucket.
-+ Admin deletes the app pod.
-+ The cosi-node-adapter is notified (_NodeUnpublishVolume_) and removes the pod.name finalizer from the BA, and the BA is garbage collected.
-+ The user BAR remains pointing to a BA that no longer exists. It is felt that COSI should not delete user-created instances.
+A Pod with a volume of type `objectstorage.k8s.io` and a pointer to the BucketAccessRequest is required for COSI to fetch information for a particular Bucket. 
```

Comment:
Is this the correct usage of "volume type"?

Reply:
Nice catch! It should be "CSI volume with driver `objectstorage...` and a volumeAttribute with key `bucketAccessRequest` pointing to the name of the BAR"